Source/Applications/TsfManager/App.xaml.cs

```diff
@@ -98,6 +101,12 @@ public Guid NodeID
 
         #region [ Methods ]
 
+        private void Application_Startup(object sender, StartupEventArgs e)
+        {
+            XmlLanguage xmlLanguage = XmlLanguage.GetLanguage(CultureInfo.CurrentCulture.IetfLanguageTag);
```

Comment:
For distribution to other TSL applications: I haven't looked at what exceptions these functions calls might throw, but should we wrap these updates in a `try / catch` then just log exception, e.g., with `Logger.SwallowException`.  Could be safe enough, but this is a critical startup area.

Reply:
Right, so looking into this, I first found this note about `IetfLanguageTag`.

> This property (and the [GetCultureInfoByIetfLanguageTag](https://learn.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.getcultureinfobyietflanguagetag?view=netframework-4.8) method) is deprecated. Instead, you should use the [CultureInfo.Name](https://learn.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.name?view=netframework-4.8) property. IETF tags and names are identical.

Neither `IetfLanguageTag` nor `CultureInfo.Name` have any documented exceptions. `XmlLanguage.GetLanguage()` does throw `ArgumentException` if the input does not conform to the syntax specified in RFC 3066. The documentation for `IetfLanguageTag` and `CultureInfo.Name` does not mention RFC 3066 so it's perhaps reasonable to assume something could go wrong in an edge case.

That said, I also tried this in PowerShell and didn't get any errors.

```powershell
Add-Type -AssemblyName PresentationCore
[Globalization.CultureInfo]::GetCultures("AllCultures") | % { [Windows.Markup.XmlLanguage]::GetLanguage($_.Name) }
```

As for `DependencyProperty.OverrideMetadata()`, it doesn't look like we should need to worry about exceptions there as it relates to overriding metadata on read-only properties and properties where metadata was "already established".

Anyway, you bring up a good point about this being a critical startup area. Even if there was provably no possibility for exceptions here, it would still be a good idea to add the try-catch.